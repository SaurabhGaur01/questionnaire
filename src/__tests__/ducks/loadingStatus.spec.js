import {
    setLoadingSuccess,
    setLoadingInProgress,
    setLoadingError,
    loadingStatusReducer,
} from '../../ducks/loadingStatus';

import { 
    LOADING_ERROR, LOADING_IN_PROGRESS, LOADING_SUCCESS
} from '../../constants/steps';

describe('loadingStatus Duck Tests', () => {
    const mockState = 'mockKey';

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setLoadingSuccess should match the snapshot' , () => {
            expect(setLoadingSuccess()).toMatchSnapshot();
        });
        it('setLoadingInProgress should match the snapshot' , () => {
            expect(setLoadingInProgress()).toMatchSnapshot();
        });
        it('setLoadingError should match the snapshot' , () => {
            expect(setLoadingError()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(loadingStatusReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(loadingStatusReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(loadingStatusReducer(mockState, unknownAction)).toBe(mockState);
        });
    });

    describe('when called with a particular action', () => {
        it('should return the value of the constant LOADING_SUCCESS' , () => {
            expect(loadingStatusReducer(mockState, setLoadingSuccess()))
            .toEqual(LOADING_SUCCESS);
        });
        it('should return the value of the constant LOADING_IN_PROGRESS' , () => {
            expect(loadingStatusReducer(mockState, setLoadingInProgress()))
            .toEqual(LOADING_IN_PROGRESS);
        });
        it('should return the value of the constant LOADING_ERROR' , () => {
            expect(loadingStatusReducer(mockState, setLoadingError()))
            .toEqual(LOADING_ERROR);
        });
    });

})