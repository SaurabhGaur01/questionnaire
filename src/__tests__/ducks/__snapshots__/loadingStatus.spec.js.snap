// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadingStatus Duck Tests Action Generator setLoadingError should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_LOADING_ERROR",
}
`;

exports[`loadingStatus Duck Tests Action Generator setLoadingInProgress should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_LOADING_IN_PROGRESS",
}
`;

exports[`loadingStatus Duck Tests Action Generator setLoadingSuccess should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_LOADING_SUCCESS",
}
`;
